### PR TITLE
expose automatic chef_environment attr

### PR DESCRIPTION
**Pull request by thommay**
We should've done this ages ago, but let's get it done.
This is also important should we wish to integrate hoist
directly in to chef.

### CODE REVIEW

**Comment by coderanger** [comment]
Very :+1: and we should probably make `node['name']` Do The Right Thing too. The only slight complication is people miiiiight misunderstand what `node.set['chef_enviroment'] = 'prod'` does (i.e. it does nothing) but I think that is 100% worth it given how many people trip up on this.

**Comment by lamont-granquist** [comment]
normal level though right?  run_list and tags are normal.

**Comment by coderanger** [comment]
I would imagine automatic, thats how we set things like `node['recipes']` and `node['policy_revision']` currently.

File: new/chef_env_attribute.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Author: Thom May <thom@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Exposing some core settings as attributes
+
+There are a number of settings, such as the node name and the environment, that we've historically not exposed as automatic attributes.
+The environment attribute is a bit more complicated in the world of
+policy groups, and this RFC clarifies that situation too
```
Comment (tas50) [inline new/chef_env_attribute.md]: Can we give an example of how you'd reference this as an attribute vs. the "old" way.

**Comment by thommay** [comment]
I was definitely thinking automatic.

**Comment by lamont-granquist** [comment]
okay so `cookbooks`, `recipes`,  `expanded_run_list` and `roles` are all automatic...

File: new/chef_env_attribute.md (hunk 1 above)
Comment (thommay) [reply new/chef_env_attribute.md]: Not sure that the audience should need that level of explanation?

**Comment by thommay** [comment]
@chef/rfc-editors we done approved this!

**Pull request merged**